src/main/java/uk/gov/companieshouse/api/accounts/service/impl/AbstractServiceImpl.java

```diff
@@ -37,6 +39,15 @@ public C save(C rest, String companyAccountId) throws NoSuchAlgorithmException {
         return rest;
     }
 
+    @Override
+    public E findByExample(E example) {
+        Optional<E> optional = (Optional<E>) mongoRepository.findOne(Example.of(example));
+        if (optional.isPresent()) {
+            return optional.get();
+        }
+        return null;
```

Comment:
Records should be retrieved out of the database using the ID. For child elements (e.g. small-full) generate the expected ID using the transaction and account ID.